eval/src/tests/ann/gist_benchmark.cpp

```diff
@@ -0,0 +1,295 @@
+// Copyright 2020 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/vespalib/testkit/test_kit.h>
+#include <vespa/vespalib/util/priority_queue.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <chrono>
+
+#define NUM_DIMS 960
+#define NUM_DOCS 200000
+#define NUM_REACH 10000
+#define NUM_Q 1000
+
+#include "doc_vector_access.h"
+#include "nns.h"
+#include "for-sift-hit.h"
+#include "for-sift-top-k.h"
+
+std::vector<TopK> bruteforceResults;
+
+struct PointVector {
```

Comment:
I don't think we should rely on this amount of copy-paste-mutate. A lot of this code is already in sift_benchmark.cpp, the only difference being the number of dimensions. Please move the following to a common place: PointVector, DocVectorAdapter, computeDistance, read_queries, read_docs, read_data, BfHitComparator, BfHitHeap. I am afraid this code will be hard to maintain otherwise.